src/Socket.cpp

```diff
@@ -699,6 +708,15 @@ int Socket::Recv(void* buffer, int length, bool block)
     return recv(socket_, reinterpret_cast<char*>(buffer), length, (block ? 0 : MSG_PEEK) );
 }
 
+int Socket::Recv(void* const buffer, const int length, sockaddr_in& addr)
+{

```

Comment:
Yes, one could create a wrapper struct for `sockaddr`. But it would (most probably) contain a `sockaddr`...

What should be possible is call a `Send` operation with what you got from the `Recv` call